File: WordPressCom-Stats-iOS/StatsItem.m
#import "StatsItem.h"

@implementation StatsItem

- (instancetype)init
{
    self = [super init];
    if (self) {
        _children = [NSMutableArray new];
    }
    return self;
}


- (NSString *)description
{
    return [NSString stringWithFormat:@"StatsItem - itemID: %@, label: %@, value: %@", self.itemID, self.label, self.value];
}


- (void)addChildStatsItem:(StatsItem *)statsItem
{
    statsItem.parent = self;
    [self.children addObject:statsItem];
}

@end

File: WordPressCom-Stats-iOS/StatsItem.swift
import Foundation

struct StatsItemAction {
    var url: NSURL?
    var label: String?
    var icon: UIImage?
    var defaultAction: Bool = false
}

class StatsItem {
    var value: NSNumber
    var label: String
    var icon: UIImage?
    var actions: Array<StatsItemAction>
    var children: Array<StatsItem>
    
    init(value: NSNumber, label: String, icon: UIImage?, actions: Array<StatsItemAction>, children: Array<StatsItem>) {
        self.value = value
        self.label = label
        self.icon = icon
        self.actions = actions
        self.children = children
    }
}